### PR TITLE
Post-Delete Unit tests for `epoll` implementation

### PR DESCRIPTION
**Pull request by mlrcbsousa**
To run tests from project root individually:

```shell
make socket -C test/unit
make parser -C test/unit
make status -C test/unit
make handler -C test/unit
make file_handler -C test/unit
make server -C test/unit
```

To run all:

```shell
make test_unit
```

### CODE REVIEW

**Review by coderMeira — COMMENTED** [review]
Sounds good

**Review by coderMeira — APPROVED** [review]

File: src/Socket.cpp [hunk 1]
```diff
@@ -132,6 +132,7 @@ void	Socket::receive( int buffer_size ) {
 
 	_bytes = recv(_fd, _buffer.data(), buffer_size, 0);
 	std::cout << "We just received " << _bytes << " bytes." << std::endl;
+	// What is this comparison for and what does it mean?
```
Comment (gleal42) [inline src/Socket.cpp:135]: Kqueue event allows us to know how many bytes the socket is ready to read everytime.
I'm not sure if epoll also has the same feature.
But basically I was counting on this number from Kqueue to be correct.
If it is not correct than Kqueue has stopped working properly hence the exception.

**Review by gleal42 — APPROVED** [review]
Great! Ready to merge!

**Comment by mlrcbsousa** [comment]
This isn't complete sorry, plan is to have a unit test file for each class file

**Comment by mlrcbsousa** [comment]
Added more unit tests for Server class, gonna go with this.

Also small adaptations to make some namings clearer, shouldn't impact main code.

**Pull request merged**